Replaced nats-operator with nats helm chart (#12415)
kyma-project/kyma: resources/eventing/charts/nats/README.md

# NATS Server

> This Helm chart deploy NATS: https://nats.io/ using NATS [Helm chart](https://github.com/nats-io/k8s/tree/v0.9.0/helm/charts/nats).

Reviewer: ```suggestion

## Overview

This Helm chart deploys [NATS](https://nats.io/) using NATS [Helm chart](https://github.com/nats-io/k8s/tree/v0.9.0/helm/charts/nats).
```

> Steps:

Reviewer: ```suggestion

## Installation

To install this chart, follow these steps:
```

> - Install NATS into "nats" namespace using Helm 3 :

Reviewer: ```suggestion
1. Install NATS into the `nats` namespace using Helm 3 :
```

```bash
kubectl create namespace nats
helm template nats nats -n nats | kubectl apply -f -
```

> - Test the installation:

Reviewer: ```suggestion
2. Test the installation:
```

```bash
$ kubectl -n nats port-forward nats-1 4222
```

## Configuration

### Limits

```yaml
nats:
  # The number of connect attempts against discovered routes.
  connectRetries: 30

  # How many seconds should pass before sending a PING
  # to a client that has no activity.
  pingInterval: 

  # Server settings.
  limits:
    maxConnections: 
    maxSubscriptions: 
    maxControlLine: 
    maxPayload: 

    writeDeadline: 
    maxPending: 
    maxPings: 
    lameDuckDuration: 

  # Number of seconds to wait for client connections to end after the pod termination is requested
  terminationGracePeriodSeconds: 60
```

### Logging

> *Note*: It is not recommended to enable trace or debug in production since enabling it will significantly degrade performance.

Reviewer: ```suggestion
> **NOTE**: It is not recommended to enable trace or debug in production, since enabling it will significantly degrade performance.
```

```yaml
nats:
  logging:
    debug: 
    trace: 
    logtime: 
    connectErrorReports: 
    reconnectErrorReports: 
```

> ## Clustering
> 
> If clustering is enabled, then a 3-node cluster will be setup. More info at:
> https://docs.nats.io/nats-server/configuration/clustering#nats-server-clustering

Reviewer: The link https://docs.nats.io/nats-server/configuration/clustering#nats-server-clustering is not working 🤔  I think it's https://docs.nats.io/legacy/stan/intro/clustering 

```suggestion
### Clustering

If clustering is enabled, then a 3-node cluster will be set up. You can find more information in the [NATS documentation](https://docs.nats.io/legacy/stan/intro/clustering).
```

```yaml
cluster:
  enabled: true
  replicas: 3
```

> ## JetStream
> 
> ### Setting up Memory and File Storage
> 

Reviewer: ```suggestion
### JetStream

To set up memory and file storage with JetStream, use this configuration: 

```

```yaml
nats:
  image: nats:alpine

  jetstream:
    enabled: true

    memStorage:
      enabled: true
      size: 2Gi

    fileStorage:
      enabled: true
      size: 1Gi
      storageDirectory: /data/
      storageClassName: default
```

> ## Misc
> 
> ### Configuration Reload sidecar
> 
> The NATS config reloader image to use:

Reviewer: ```suggestion
### Reload sidecar

To configure the reload sidecar, use this NATS config reloader image: 
```
